### PR TITLE
reorder everything

### LINKED ISSUE #224: Reorganize init arguments
Some things are bad... for example, rel_path of images is like 3rd or 4th when it should be first. We need to go through everything and make sure that it's good.

### PR DESCRIPTION
**Pull request by tinmarr**
Reorders parameters

Fixes #224 

@tsedan and @mrElnekave please both approve before merging.
## Hello reviewers!
Before approving this PR, please ensure the following:

  - [x] Documentation is up to date
  - [x] Relevent tutorials are up to date
  - [x] Changelog is up to date
  - [x] Test coverage is at 100%
  - [x] All checks are passing
  - [x] That the new code actually works

Once all this is done, you can be sure that the PR is ready to be merged. 🎉

*This message has been added automatically. Please don't delete it.*

### CODE REVIEW

File: rubato/struct/gameobject/raster.py [hunk 1]
```diff
@@ -12,20 +12,20 @@ class Raster(Component):
...
     def __init__(
         self,
-        offset: Vector = Vector(),
-        rot_offset: float = 0,
         width: int = 32,
         height: int = 32,
         scale: Vector = Vector(1, 1),
+        offset: Vector = Vector(),
+        rot_offset: float = 0,
         z_index: int = 0
```
Comment (mrElnekave) [inline rubato/struct/gameobject/raster.py:29]: I believe z_index comes before offsets for GOs can we keep this pattern?

File: rubato/struct/scene.py [hunk 2]
```diff
@@ -28,9 +28,9 @@ class Scene:
 
     def __init__(
         self,
-        name: str = "default",
+        background_color: Color = Color(255, 255, 255),
         border_color: Color = Color(),
-        background_color: Color = Color(255, 255, 255)
+        name: str = "default",
```
Comment (mrElnekave) [inline rubato/struct/scene.py:33]: Is there a reason name isn't first anymore. `intro_scene = Scene("intro")` I believe is important.

**Review by mrElnekave — APPROVED** [review]
Looks good, this was a long time overdue. See comment.

**Comment by mrElnekave** [comment]
@tsedan Could u check if the tutorials work lol.

**Comment by tinmarr** [comment]
Tutorials will get fixed when #222 gets fixed

**Review by tsedan — APPROVED** [review]
looks good. will need to revisit imports after merge because stuff is broken, likely from when we rearranged the struct folder.

**Pull request merged**